Libs/MRML/Logic/vtkMRMLSliceLogic.cxx

```diff
@@ -1461,31 +1527,29 @@ void vtkMRMLSliceLogic::CreateSliceModel()
 
 //----------------------------------------------------------------------------
 vtkMRMLVolumeNode *vtkMRMLSliceLogic::GetLayerVolumeNode(int layer)
+{
+  vtkWarningMacro("Deprecated, please use GetNthLayerVolumeNode instead");
+  return this->GetNthLayerVolumeNode(layer);
+}
+
+//----------------------------------------------------------------------------
+vtkMRMLVolumeNode *vtkMRMLSliceLogic::GetNthLayerVolumeNode(int layer)
 {
   if (!this->SliceNode || !this->SliceCompositeNode)
   {
     return (nullptr);
   }
-
-  const char *id = nullptr;
-  switch (layer)
+  if (layer < 0)
   {
-    case LayerBackground:
-    {
-      id = this->SliceCompositeNode->GetBackgroundVolumeID();
-      break;
-    }
-    case LayerForeground:
-    {
-      id = this->SliceCompositeNode->GetForegroundVolumeID();
-      break;
-    }
-    case LayerLabel:
-    {
-      id = this->SliceCompositeNode->GetLabelVolumeID();
-      break;
-    }
+    vtkErrorMacro(<< "GetNthLayerVolumeNode: Non-negative layer index is expected.");
+    return nullptr;
+  }
+  int maxLayerIndex = vtkMRMLSliceLogic::Layer_Last;
+  if (layer >= maxLayerIndex)
+  {
+    return nullptr;
   }
+  const char *id = this->SliceCompositeNode->GetNthLayerVolumeID(layer);
```

Comment:
As indicated by the failed build results, `GetNthLayerVolumeID` is unavailable here and is introduced in `ENH: Generalize CompositeNode API introducing "Nth Layer" functions` of https://github.com/Slicer/Slicer/pull/8277. This appears to be an accidental inclusion of future API in the first commit `ENH: Generalize SliceLogic API introducing "Nth Layer" functions`.

Reply:
Thanks for the review and for looking into this.

I revisited the patches so that the use of `vtkMRMLSliceCompositeNode::GetNthLayerVolumeID` only occur in commit `Generalize CompositeNode API introducing "Nth Layer" functions`